# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/ActorsParent.cpp
# Commit: 2ed5e139d784
# Full Hash: 2ed5e139d7840692dad5bcf39348e92bf3b05fdb
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-12-05 09:38:58
# Regressor Bug: 1678030
# File Overlap Count: 1
# Description:
#   Bug 1678030 - Move CreateAndExecuteSingleStepStatement to QuotaCommon and use in all quota clients. r=dom-workers-and-storage-reviewers,janv
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D97460
# ==============================================================================

diff -r 80442bbcf255 -r 2ed5e139d784 dom/quota/ActorsParent.cpp
--- a/dom/quota/ActorsParent.cpp	Fri Dec 04 15:08:17 2020 +0000
+++ b/dom/quota/ActorsParent.cpp	Fri Dec 04 14:56:18 2020 +0000
@@ -6029,20 +6029,9 @@
 
   if (connection) {
     // Find out the journal mode.
-    nsCOMPtr<mozIStorageStatement> stmt;
-    rv = connection->CreateStatement("PRAGMA journal_mode;"_ns,
-                                     getter_AddRefs(stmt));
-    if (NS_WARN_IF(NS_FAILED(rv))) {
-      return rv;
-    }
-
-    bool hasResult;
-    rv = stmt->ExecuteStep(&hasResult);
-    if (NS_WARN_IF(NS_FAILED(rv))) {
-      return rv;
-    }
-
-    MOZ_ASSERT(hasResult);
+    QM_TRY_INSPECT(const auto& stmt,
+                   CreateAndExecuteSingleStepStatement(
+                       *connection, "PRAGMA journal_mode;"_ns));
 
     nsCString journalMode;
     rv = stmt->GetUTF8String(0, journalMode);